node/src/accountant/big_int_db_processor.rs

```diff
@@ -0,0 +1,1613 @@
+                                None,
+                                SQLParamsBuilder::default()
+                                    .key("name", ":name", &"Joe")
+                                    .wei_change(wei_change.clone())
+                                    .build(),
+                            ),
+            );
+            tx.commit().unwrap();
+            result
+        };
+
+        precise_upsert_or_update_assertion_test_environment(
+            "update_alone_works_also_for_transaction_instead_of_connection",
+            Some((&"Joe", 47, 598745133)),
+            &wei_change,
+            false,
+            act,
+        )
+    }
+
+    #[test]
+    fn select_stm_in_update_with_overflow_gets_also_well_along_numeric_key_value_and_different_table_and_substitution_param_name(
+    ) {
+        let conn = create_new_empty_db(
+            "big_int_db_processor",
+            "select_stm_in_update_with_overflow_gets_also_well_along_numeric_key_value_and_different_table_and_substitution_param_name"
+        );
+        let conn = ConnectionWrapperReal::new(conn);
+        let mut stm = conn.prepare("create table test_table \
+        (family_members_count int primary key, costs_per_month_per_district_high_b int not null, costs_per_month_per_district_low_b int not null)").unwrap();
+        stm.execute([]).unwrap();
+        let mut stm = conn.prepare("insert into test_table \
+        (family_members_count, costs_per_month_per_district_high_b, costs_per_month_per_district_low_b) values (4,4578,5468956)").unwrap();
+        stm.execute([]).unwrap();
+        let balance_change = Addition("costs_per_month_per_district", 50000);
+        let update_config = BigIntSqlConfig::new(
+            "update test_table set costs_per_month_per_district_high_b = costs_per_month_per_district_high_b + :costs_per_month_per_district_high_b,\
+             costs_per_month_per_district_low_b = costs_per_month_per_district_low_b + :costs_per_month_per_district_low_b where family_members_count = :members_count",
+            None,
+            SQLParamsBuilder::default()
+                .wei_change(balance_change)
+                .key("family_members_count", ":members_count", &4)
+                .build(),
+        );
+
+        let result = BigIntDbProcessor::<DummyDao>::default()
+            .update_threatened_by_overflow(Either::Left(&conn), update_config);
+
+        assert_eq!(result, Ok(()));
+        let (high_bytes_added, low_bytes_added) = BigIntDivider::deconstruct(50000);
+        conn.prepare("select * from test_table")
+            .unwrap()
+            .query_row([], |row| {
+                let member_count = row.get::<usize, i64>(0).unwrap();
+                let high_bytes = row.get::<usize, i64>(1).unwrap();
+                let low_bytes = row.get::<usize, i64>(2).unwrap();
+                assert_eq!(member_count, 4);
+                //the added 1 seems wrong, but we're exercising code intended
```

Comment:
Yes, this _does_ seem wrong. I understand it right this second, but three seconds from now I might forget and not understand it again. Let's see if I can do any better. Hmm.

// This test verifies that the overflow-handling code correctly adds 1 to the high end of the 128-bit number. The scenario in the test doesn't justify the overflow handling, but we still have to verify that it happens.

I don't know whether that's better, but right at this moment I understand it more. That may fade. I don't know. Another way to deal with it might be to write the test such that the scenario produced an overflow.